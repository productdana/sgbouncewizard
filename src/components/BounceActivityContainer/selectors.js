import {
  selectSelectorGenerator,
  writeSelectorGenerator,
} from "../selectorHelpers";

// These test hooks will eventually turn into the following:
// - Selectors (CSS Selector strings) i.e. "[data-test=emailInput]"
// - Write Selectors (Selector objects spread onto React components) i.e. { data-test: "emailInput" }
// What if we had to update a data-test attribute value or other CSS selectors?
// - We would have to update all the test file strings in Jest/Enzyme and Cypress tests and React components.
// - This pattern helps us maintain this in the long run and update as few places as possible for CSS selector changes
const testHooks = {
  page: "activity-log-page",
  breadcrumb: "activity-log-breadcrumb",
  csvButton: "export-csv-button",
  activityFilter: "rule-filter",
  activityTable: "activity-table",
  emptyRulesWarning: "empty-rules-warning",
  pagination: "pagination",
  detailsModal: "actvity-details-modal",
};

// Selectors like "[data-test=emailInput]" to be imported into the following files:
// - Jest/Enzyme tests to be placed into .find("[data-test=emailInput]") and other similar functions
// - Cypress tests to be placed into cy.get("[data-test=emailInput]"), cy.contains(), etc.
const Selectors = selectSelectorGenerator(testHooks);

// Write Selectors like { emailInput: { data-test: "emailInput" }, ... } to be imported into
// React component files to spread the data-test attributes onto elements we would like to keep track of for tests
// i.e. <Component {...WriteSelectors.emailInput} /> => <Component data-test="emailInput" />
const WriteSelectors = writeSelectorGenerator(testHooks);

export { Selectors, WriteSelectors };
