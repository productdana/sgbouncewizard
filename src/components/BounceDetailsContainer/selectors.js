import {
  selectSelectorGenerator,
  writeSelectorGenerator,
} from "../selectorHelpers";

// These test hooks will eventually turn into the following:
// - Selectors (CSS Selector strings) i.e. "[data-test=emailInput]"
// - Write Selectors (Selector objects spread onto React components) i.e. { data-test: "emailInput" }
// What if we had to update a data-test attribute value or other CSS selectors?
// - We would have to update all the test file strings in Jest/Enzyme and Cypress tests and React components.
// - This pattern helps us maintain this in the long run and update as few places as possible for CSS selector changes
const testHooks = {
  details: "details",
  detailsEditable: "details-editable",
  changelog: "changelog",
  editButton: "edit-button",
  saveButton: "save-button",
  cancelButton: "cancel-button",
  description: "description",
  bounceId: "bounce_id",
  responseCode: "response-code",
  enhancedCode: "enhanced-code",
  regex: "regex",
  priority: "priority",
  bounceAction: "bounce-action",
  changelogModal: "changelog-modal",
  confirmModal: "commit-confirm-modal",
  saveConfirmationModal: "save-confirmation-modal",
  cancelConfirmationModal: "cancel-confirmation-modal",
  confirmationSubmit: "create-confirm-submit",
  commitInput: "commit-input",
  confirmSubmit: "confirm-submit",
  pagination: "pagination",
};

// Selectors like "[data-test=emailInput]" to be imported into the following files:
// - Jest/Enzyme tests to be placed into .find("[data-test=emailInput]") and other similar functions
// - Cypress tests to be placed into cy.get("[data-test=emailInput]"), cy.contains(), etc.
const Selectors = selectSelectorGenerator(testHooks);

// Write Selectors like { emailInput: { data-test: "emailInput" }, ... } to be imported into
// React component files to spread the data-test attributes onto elements we would like to keep track of for tests
// i.e. <Component {...WriteSelectors.emailInput} /> => <Component data-test="emailInput" />
const WriteSelectors = writeSelectorGenerator(testHooks);

export { Selectors, WriteSelectors };
