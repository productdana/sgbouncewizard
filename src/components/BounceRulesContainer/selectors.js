import {
  selectSelectorGenerator,
  writeSelectorGenerator,
} from "../selectorHelpers";

// These test hooks will eventually turn into the following:
// - Selectors (CSS Selector strings) i.e. "[data-test=emailInput]"
// - Write Selectors (Selector objects spread onto React components) i.e. { data-test: "emailInput" }
// What if we had to update a data-test attribute value or other CSS selectors?
// - We would have to update all the test file strings in Jest/Enzyme and Cypress tests and React components.
// - This pattern helps us maintain this in the long run and update as few places as possible for CSS selector changes
const testHooks = {
  page: "bounce-rules-page",
  breadcrumb: "bounce-rules-breadcrumb",
  csvButton: "export-csv-button",
  createRuleButton: "create-rule-button",
  cancelCreateRuleButton: "cancel-create-rule-button",
  bounceRule: "bounce-rule",
  ruleFilter: "rule-filter",
  ruleTable: "rule-table",
  emptyRulesWarning: "empty-rules-warning",
  deleteConfirmation: "delete-confirmation",
  deleteConfirmationConfirm: "delete-confirmation-confirm",
  deleteAlert: "delete-alert",
  deleteButton: "delete-button",
  createRuleModal: "create-rule-modal",
  priority: "create-priority",
  description: "create-description",
  bounceAction: "create-bounce-action",
  responseCode: "create-response-code",
  enhancedCode: "create-enhanced-code",
  regex: "create-regex",
  submitButton: "create-submit-button",
  confirmModal: "commit-confirm-modal",
  confirmationSubmit: "create-confirm-submit",
  commitInput: "commit-message",
  cancelConfirmationSubmit: "cancel-confirm-submit",
  emptyCommitAlert: "empty-commit-alert",
  pagination: "pagination",
  confirmSubmit: "confirm-submit",
  invalidInput: "invalid-input",
};

// Selectors like "[data-test=emailInput]" to be imported into the following files:
// - Jest/Enzyme tests to be placed into .find("[data-test=emailInput]") and other similar functions
// - Cypress tests to be placed into cy.get("[data-test=emailInput]"), cy.contains(), etc.
const Selectors = selectSelectorGenerator(testHooks);

// Write Selectors like { emailInput: { data-test: "emailInput" }, ... } to be imported into
// React component files to spread the data-test attributes onto elements we would like to keep track of for tests
// i.e. <Component {...WriteSelectors.emailInput} /> => <Component data-test="emailInput" />
const WriteSelectors = writeSelectorGenerator(testHooks);

export { Selectors, WriteSelectors };
