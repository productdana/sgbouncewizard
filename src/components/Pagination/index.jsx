import React from "react";
import cn from "classnames";
import { Column } from "../Column";
import "./index.scss";

function lastPageToDisplay(currentPageIndex, pagesToDisplay, totalPages) {
  const lastPageCalculation =
    currentPageIndex +
    (pagesToDisplay -
      (currentPageIndex % pagesToDisplay === 0
        ? pagesToDisplay
        : currentPageIndex % pagesToDisplay));
  const lastPage =
    lastPageCalculation > totalPages ? totalPages : lastPageCalculation;
  return lastPage;
}

function firstPageToDisplay(endPage, pagesToDisplay) {
  const startPage = endPage - (pagesToDisplay - 1);
  return startPage <= 1 ? 1 : startPage;
}

function calcShouldDisplayLastPage(
  totalPages,
  pagesToDisplay,
  currentPageIndex,
  endPage
) {
  if (totalPages < pagesToDisplay) {
    return false;
  }
  if (currentPageIndex <= endPage) {
    return true;
  }
  return false;
}

function showPaginationLinks(
  startPage,
  endPage,
  currentPageIndex,
  updatePageIndex
) {
  const numberOfPages = endPage - startPage + 1;

  return Array(numberOfPages)
    .fill()
    .map((_, i) => {
      const currentPaginationNum = startPage + i;
      return (
        <a
          key={currentPaginationNum}
          className={cn("pagination-link", {
            "is-active": currentPaginationNum === currentPageIndex,
          })}
          onClick={updatePageIndex}
          onKeyDown={updatePageIndex}
          value={currentPaginationNum}
          role="button"
          tabIndex="0"
        >
          {currentPaginationNum}
        </a>
      );
    });
}

const Pagination = ({
  handlePrevClicked,
  handleNextClicked,
  updatePageIndex,
  currentPageIndex,
  rulesToShow,
  pagesToDisplay,
  numRules,
}) => {
  const totalPages = Math.ceil(numRules / rulesToShow);
  const endPage = lastPageToDisplay(
    currentPageIndex,
    pagesToDisplay,
    totalPages
  );
  const startPage = firstPageToDisplay(endPage, pagesToDisplay);
  const shouldDisplayFirstPage = currentPageIndex > pagesToDisplay;
  const shouldDisplayLastPage = calcShouldDisplayLastPage(
    totalPages,
    pagesToDisplay,
    currentPageIndex,
    endPage
  );
  const shouldDisplayPrev = currentPageIndex === 1;
  const shouldDisplayNext = currentPageIndex === totalPages;
  return (
    <Column width={10} offset={2}>
      <div className="pagination pagination-container">
        <a
          className={`btn btn-secondary btn-small pagination-prev ${
            shouldDisplayPrev ? "is-invisible" : ""
          }`}
          onClick={handlePrevClicked}
          onKeyDown={handlePrevClicked}
          role="button"
          tabIndex="0"
        >
          Prev
        </a>
        {shouldDisplayFirstPage && (
          <span>
            <a
              className="pagination-link"
              onClick={updatePageIndex}
              onKeyDown={updatePageIndex}
              value={1}
              role="button"
              tabIndex="0"
            >
              {1}
            </a>
            <a className="pagination-ellipses">&hellip;</a>
          </span>
        )}
        <div className="pagination-links">
          {showPaginationLinks(
            startPage,
            endPage,
            currentPageIndex,
            updatePageIndex
          )}
          {/* {Array(endPage - startPage + 1)
            .fill()
            .map((_, i) => startPage + i)
            .map(number => (
              <a
                key={number}
                className={cn("pagination-link", {
                  "is-active": number === currentPageIndex
                })}
                onClick={updatePageIndex}
                onKeyDown={updatePageIndex}
                value={number}
                role="button"
                tabIndex="0"
              >
                {number}
              </a>
            ))} */}
          {shouldDisplayLastPage && (
            <span>
              <a className="pagination-ellipses">&hellip;</a>
              <a
                className="pagination-link"
                onClick={updatePageIndex}
                onKeyDown={updatePageIndex}
                value={totalPages}
                role="button"
                tabIndex="0"
              >
                {totalPages}
              </a>
            </span>
          )}
        </div>
        <a
          className={`btn btn-secondary btn-small pagination-next ${
            shouldDisplayNext ? "is-invisible" : ""
          }`}
          onClick={handleNextClicked}
          onKeyDown={handleNextClicked}
          role="button"
          tabIndex="0"
        >
          Next
        </a>
      </div>
    </Column>
  );
};

export default Pagination;
