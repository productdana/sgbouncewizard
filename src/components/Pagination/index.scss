.pagination-container {
  margin-bottom: 30px;
}

.is-invisible {
  display: none;
}
