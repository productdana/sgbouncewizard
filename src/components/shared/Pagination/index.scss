.pagination-container {
  margin-bottom: 30px;
}

.is-invisible {
  visibility: hidden;
}
