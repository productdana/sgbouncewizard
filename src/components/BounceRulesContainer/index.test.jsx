import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { shallow } from "enzyme";
import BounceRuleContainer from ".";
import Pagination from "../shared/Pagination";
import { Selectors } from "./selectors";

const {
  csvButton,
  createRuleButton,
  ruleFilter,
  ruleTable,
  emptyRulesWarning,
  createRuleModal,
  confirmModal,
} = Selectors;

const testRules = [
  {
    id: 504,
    response_code: 550,
    enhanced_code: "",
    regex: "no MX record for domain",
    priority: 0,
    description:
      "bWFpbmx5IGxpYmVydHkgZG9tYWluIGJsb2NrIHNlZWluZyB+NTAlIG9mIGFkZHJlc3NlcyBlbmdhZ2luZyBTRyB3aWRl",
    bounce_action: "no_action",
  },
];

const testActivity = [
  {
    id: 173,
    response_code: 421,
    enhanced_code: "",
    regex: "",
    priority: 0,
    description: "RFC5321Servicenotavailable",
    bounce_action: "retry",
    user_id: 1,
    created_at: 1,
    comment: "Inital Setup",
    operation: "New",
  },
];

const defaultProps = {
  rules: testRules,
  filteredRules: testRules,
  filterOptions: [],
  pageIndex: 1,
  pagesToDisplay: 5,
  activityLog: testActivity,
  filteredActivityLog: testActivity,
  isFetching: false,
  isBounceRulesTab: true,
};

const wrapper = shallow(<BounceRuleContainer {...defaultProps} />);

it("should render correctly", () => {
  const tree = renderer
    .create(
      <Router>
        <BounceRuleContainer {...defaultProps} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("should render a create a bounce rule", () => {
  expect(wrapper.find(createRuleButton)).toHaveLength(1);
});

it("should render a export to csv button", () => {
  expect(wrapper.find(csvButton)).toHaveLength(1);
});

it("should render a filter component", () => {
  expect(wrapper.find(ruleFilter)).toHaveLength(1);
});

it("should render a rule list component", () => {
  expect(wrapper.find(ruleTable)).toHaveLength(1);
});

it("should render paginiation", () => {
  expect(wrapper.find(Pagination)).toHaveLength(1);
});

it("should render warning when no rules available", () => {
  wrapper.setProps({ filteredRules: [], rules: [] });
  expect(wrapper.find(emptyRulesWarning)).toHaveLength(1);
});

it("should render create rule modal", () => {
  wrapper.setProps({ isCreateRuleOpen: true });
  expect(wrapper.find(createRuleModal)).toHaveLength(1);
});

it("should render create rule confirmation", () => {
  wrapper.setProps({ isCreateRuleConfirmationOpen: true });
  expect(wrapper.find(confirmModal).exists()).toBeTruthy();
});
