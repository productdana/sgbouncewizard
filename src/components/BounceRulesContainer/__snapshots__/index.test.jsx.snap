// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="container"
  data-test="bounce-rules-page"
>
  <div
    className="header-container"
  >
    <div
      className="mako-grid__row"
    >
      <div
        className="mako-grid__col-3 col-start-2"
      >
        <img
          alt="sendgrid-logo"
          className="sg-logo"
          data-test="sg-logo"
          src="https://uiux.s3.amazonaws.com/toggleable-logos/header-logo.svg"
          style={
            Object {
              "maxWidth": "150px",
            }
          }
        />
      </div>
      <div
        className="mako-grid__col-2 col-start-10"
      >
        <div
          className="header-user-container"
        >
          <span
            className="header-greeting"
            data-test="user-greeting"
          >
            Hello, 
            User
            !
          </span>
          <a
            className="header-logout"
            data-test="logout-button"
            role="button"
            tabIndex={0}
          >
            Logout
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="mako-grid__row"
  >
    <div
      className="mako-grid__col-6 col-start-2"
    >
      <ol
        className="breadcrumb"
      >
        <li>
          <a
            href="/bounce_rules"
            onClick={[Function]}
          >
             Bounce Rules
          </a>
        </li>
        <li />
      </ol>
    </div>
  </div>
  <div
    className="mako-grid__row"
  >
    <div
      className="mako-grid__col-4 col-start-8 csv-button-col"
    >
      <a
        className="sg-button btn btn-secondary sg-right"
        data-test="export-csv-button"
        download="bounce_rules.csv"
        href="data:text/csv;charset=utf-8,ï»¿\\"id\\",\\"response_code\\",\\"enhanced_code\\",\\"regex\\",\\"priority\\",\\"description\\",\\"bounce_action\\"
\\"504\\",\\"550\\",\\"\\",\\"no MX record for domain\\",\\"0\\",\\"bWFpbmx5IGxpYmVydHkgZG9tYWluIGJsb2NrIHNlZWluZyB+NTAlIG9mIGFkZHJlc3NlcyBlbmdhZ2luZyBTRyB3aWRl\\",\\"no_action\\""
        onClick={[Function]}
        target="_blank"
      >
        Export CSV
      </a>
      <button
        className="btn btn-primary create-rule-button"
        data-button="create-button"
        data-test="create-rule-button"
        disabled={false}
        id="isCreateRuleOpen"
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        Create Rule
      </button>
    </div>
  </div>
  <div
    className="mako-grid__row"
  >
    <div
      className="mako-grid__col-10 col-start-2"
    >
      <ul
        className="tab-wrapper page-tab"
      >
        <li
          className="tab undefined is-active"
          onClick={[Function]}
        >
          Bounce Rules
           
        </li>
        <li
          className="tab"
          onClick={[Function]}
        >
          Activity Log
           
        </li>
      </ul>
    </div>
  </div>
  <div
    className="mako-grid__row"
  >
    <div
      className="mako-grid__col-10 col-start-2"
    >
      <div
        data-test="rule-filter"
      >
        <div
          className="filter-wrap"
        >
          <div
            className="filter-header"
          >
            <p
              className="filter-title"
            >
              Search bounce rules by 
              <a
                className="filter-option-toggle"
                href="#"
              >
                all
              </a>
               of the following:
            </p>
            <div
              className="btn-list"
            >
              <button
                className="btn btn-secondary btn-small"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Clear Search
              </button>
            </div>
          </div>
          <div
            className="filter-list"
          >
            <div
              className="mako-grid__row"
            >
              <div
                className="mako-grid__col-3 col-start-1"
              >
                <div
                  className="input-select-wrap"
                >
                  <div
                    className="css-10nd86i select-filter"
                    onKeyDown={[Function]}
                  >
                    <div
                      className="css-12nb859"
                      onMouseDown={[Function]}
                      onTouchEnd={[Function]}
                    >
                      <div
                        className="css-1m8oe7c"
                      >
                        <div
                          className="css-1v8euj5"
                        >
                          Bounce Action
                        </div>
                        <div
                          className="css-1g6gooi"
                        >
                          <div
                            className=""
                            style={
                              Object {
                                "display": "inline-block",
                              }
                            }
                          >
                            <input
                              aria-autocomplete="list"
                              autoCapitalize="none"
                              autoComplete="off"
                              autoCorrect="off"
                              disabled={false}
                              id="react-select-2-input"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              spellCheck="false"
                              style={
                                Object {
                                  "background": 0,
                                  "border": 0,
                                  "boxSizing": "content-box",
                                  "color": "inherit",
                                  "fontSize": "inherit",
                                  "opacity": 1,
                                  "outline": 0,
                                  "padding": 0,
                                  "width": "1px",
                                }
                              }
                              tabIndex="0"
                              theme={
                                Object {
                                  "borderRadius": 4,
                                  "colors": Object {
                                    "danger": "#DE350B",
                                    "dangerLight": "#FFBDAD",
                                    "neutral0": "hsl(0, 0%, 100%)",
                                    "neutral10": "hsl(0, 0%, 90%)",
                                    "neutral20": "hsl(0, 0%, 80%)",
                                    "neutral30": "hsl(0, 0%, 70%)",
                                    "neutral40": "hsl(0, 0%, 60%)",
                                    "neutral5": "hsl(0, 0%, 95%)",
                                    "neutral50": "hsl(0, 0%, 50%)",
                                    "neutral60": "hsl(0, 0%, 40%)",
                                    "neutral70": "hsl(0, 0%, 30%)",
                                    "neutral80": "hsl(0, 0%, 20%)",
                                    "neutral90": "hsl(0, 0%, 10%)",
                                    "primary": "#2684FF",
                                    "primary25": "#DEEBFF",
                                    "primary50": "#B2D4FF",
                                    "primary75": "#4C9AFF",
                                  },
                                  "spacing": Object {
                                    "baseUnit": 4,
                                    "controlHeight": 38,
                                    "menuGutter": 8,
                                  },
                                }
                              }
                              type="text"
                              value=""
                            />
                            <div
                              style={
                                Object {
                                  "height": 0,
                                  "left": 0,
                                  "overflow": "scroll",
                                  "position": "absolute",
                                  "top": 0,
                                  "visibility": "hidden",
                                  "whiteSpace": "pre",
                                }
                              }
                            >
                              
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        className="css-1wy0on6"
                      >
                        <span
                          className="css-0"
                        />
                        <div
                          aria-hidden="true"
                          className="css-86sq3t"
                          onMouseDown={[Function]}
                          onTouchEnd={[Function]}
                        >
                          <svg
                            aria-hidden="true"
                            className="css-19bqh2r"
                            focusable="false"
                            height={20}
                            viewBox="0 0 20 20"
                            width={20}
                          >
                            <path
                              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="primary-filter"
                      type="hidden"
                      value="Bounce Action"
                    />
                  </div>
                </div>
              </div>
              <div
                className="mako-grid__col-9 col-start-4"
              >
                <div
                  className="input-text-wrap"
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                >
                  <label
                    className="input-text-label"
                  >
                    Description
                  </label>
                  <input
                    aria-describedby={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    step={null}
                    type="text"
                    value=""
                  />
                  
                </div>
              </div>
              <i
                className="filter-row-remove sg-icon sg-icon-x"
              />
            </div>
            <div
              className="mako-grid__row"
            >
              <div
                className="row btn-list filter-actions-container"
              >
                <button
                  className="btn btn-primary btn-small"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Add a Filter
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="mako-grid__row"
  >
    <div
      className="mako-grid__col-10 col-start-2"
    >
      <div
        data-test="rule-table"
      >
        <table
          className="table-wrap"
        >
          <thead>
            <tr>
              <th
                className="row-id"
                onClick={[Function]}
              >
                Id
              </th>
              <th
                className="row-bounce-action"
                onClick={[Function]}
              >
                Bounce Action
              </th>
              <th
                className="row-response-code"
                onClick={[Function]}
              >
                Response Code
              </th>
              <th
                className="row-description"
                onClick={[Function]}
              >
                Description
              </th>
              <th
                className="actions-cell row-action"
                onClick={[Function]}
              >
                Actions
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              data-cypress="no_action"
            >
              <td>
                504
              </td>
              <td>
                no_action
              </td>
              <td>
                550
              </td>
              <td>
                bWFpbmx5IGxpYmVydHkgZG9tYWluIGJsb2NrIHNlZWluZyB+NTAlIG9mIGFkZHJlc3NlcyBlbmdhZ2luZyBTRyB3aWRl
              </td>
              <td
                className="actions"
              >
                <i
                  className="sg-icon sg-icon-ellipsis"
                  style={null}
                />
                <div
                  className="action-icons"
                >
                  <span
                    data-tooltip="View"
                    data-tooltip-pos="up"
                    id="isRedirectingToDetail"
                    onClick={[Function]}
                    rule={504}
                  >
                    <i
                      className="sg-icon sg-icon-view"
                      style={null}
                    />
                  </span>
                  <span
                    data-rule={504}
                    data-tooltip="Delete"
                    data-tooltip-pos="up"
                    id="isDeleteConfirmationOpen"
                    onClick={[Function]}
                    rule={504}
                  >
                    <i
                      className="sg-icon sg-icon-trash"
                      style={null}
                    />
                  </span>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div
    className="mako-grid__row"
  >
    <div
      className="mako-grid__col-10 col-start-2"
    >
      <div
        className="pagination pagination-container"
      >
        <a
          className="btn btn-secondary btn-small pagination-prev is-invisible"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex="0"
        >
          Prev
        </a>
        <div
          className="pagination-links"
        />
        <a
          className="btn btn-secondary btn-small pagination-next "
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex="0"
        >
          Next
        </a>
      </div>
    </div>
  </div>
</div>
`;
